vwifi.c

```diff
@@ -53,38 +59,80 @@ static inline struct owl_ndev_priv_context *ndev_get_owl_context(
     return (struct owl_ndev_priv_context *) netdev_priv(ndev);
 }
 
-/* Helper function that will prepare structure with "dummy" BSS information and
- * "inform" the kernel about "new" BSS
- */
-static void inform_dummy_bss(struct owl_context *owl)
+static s32 rand_int(s32 low, s32 up)
 {
-    struct cfg80211_bss *bss = NULL;
-    struct cfg80211_inform_bss data = {
-        /* the only channel */
-        .chan = &owl->wiphy->bands[NL80211_BAND_2GHZ]->channels[0],
-
-        /* signal "type" not specified in this module, so it is unused.
-         * It can be some kind of percentage from 0 to 100 or mBm value.
-         * signal "type" may be specified before wiphy registration by setting
-         * wiphy->signal_type
-         */
-        .scan_width = NL80211_BSS_CHAN_WIDTH_20,
-        .signal = 1337,
-    };
+    s32 result = (s32) get_random_u32();
+    result %= (up - low + 1);
+    result = abs(result);
+    result += low;
+    return result;
+}
 
-    /* array of tags that retrieved from beacon frame or probe responce */
-    char ie[SSID_DUMMY_SIZE + 2] = {WLAN_EID_SSID, SSID_DUMMY_SIZE};
-    memcpy(ie + 2, SSID_DUMMY, SSID_DUMMY_SIZE);
+/* Helper function for generating BSSID from SSID */
+static void generate_bssid_with_ssid(u8 *result, const char *ssid)
+{
+    /*
+        Hash SSID using MurmurHash.
+        See https://stackoverflow.com/a/57960443
+    */
+    uint64_t h = 525201411107845655ull;
+    for (; *ssid; ++ssid) {
+        h ^= *ssid;
+        h *= 0x5bd1e9955bd1e995;
+        h ^= h >> 47;
+    }
 
-    /* It is posible to use cfg80211_inform_bss() instead. */
-    bss = cfg80211_inform_bss_data(
-        owl->wiphy, &data, CFG80211_BSS_FTYPE_UNKNOWN, fake_bssid, 0,
-        WLAN_CAPABILITY_ESS, 100, ie, sizeof(ie), GFP_KERNEL);
+    u64_to_ether_addr(h, result);
+    result[0] &= 0xfe; /* clear multicast bit */
+    result[0] |= 0x02; /* set local assignment bit */
+}
 
-    /* cfg80211_inform_bss_data() returns cfg80211_bss structure referefence
-     * counter of which should be decremented if it is unused.
-     */
-    cfg80211_put_bss(owl->wiphy, bss);
+/* Helper function that will prepare structure with self-defined BSS information
+ * and "inform" the kernel about "new" BSS Most of the code are copied from the
+ * upcoming inform_dummy_bss function.
+ */
+static void inform_dummy_bss(struct owl_context *owl)
+{
+    const char *org_ssid_list = (ssid_list != NULL && strlen(ssid_list))
```

Comment:
Change from `ssid_list != NULL` to `ssid_list`.

Reply:
fixed in https://github.com/sysprog21/vwifi/pull/4/commits/a02d3d9611e41eb07f7b7a401bffd387324ad764